src/coreclr/tools/Common/TypeSystem/Common/MetadataVirtualMethodAlgorithm.cs

```diff
@@ -1003,11 +1003,28 @@ private static MethodDesc TryResolveVirtualStaticMethodOnThisType(MetadataType c
                 if (methodImpl.Decl == interfaceMethodDefinition)
                 {
                     MethodDesc resolvedMethodImpl = methodImpl.Body;
-                    if (interfaceMethod != interfaceMethodDefinition)
+                    TypeDesc bodyOwningType = resolvedMethodImpl.OwningType;
+                    TypeDesc checkOwningType = constrainedType;
+                    bool isValidMethodImpl = (bodyOwningType == checkOwningType);
+                    if (!isValidMethodImpl)
+                    {
+                        while ((checkOwningType = checkOwningType.BaseType) != null)
```

Comment:
I think this is still going to reject valid things if the base type is generic (we'd need to uninstantiate).

Reply:
Upon reflection, I've thought about this a bunch, and I believe that we don't need to do this validation. As Jan comments we don't attempt to ensure that invalid IL behaves correctly in R2R builds, although we do have some amount of testing in the coreclr portion of the test suite. If we can simply disable this test we should be good to go.